src/greedy.jl

```diff
@@ -169,76 +207,81 @@ function greedy(instance::Instance)
                     tie_break[i] += Int(xor( cond1 , cond2 ))
                 end
             end
 
             # Compute the new candidate list
-            # TODO use popfirst and push to filter candidates instead of copying the table
-            tmp_candidates = copy(candidates)
-            candidates = [tmp_candidates[1]]
-            max_tie_break = tie_break[1]
-            for i in 2:length(tmp_candidates)
-              if tie_break[i] > max_tie_break
-                  candidates = [tmp_candidates[i]]
-                  max_tie_break = tie_break[i]
-              elseif tie_break[i] == max_tie_break
-                  push!(candidates, tmp_candidates[i])
-              end
+            candidates = filter_on_max_criterion(candidates, tie_break)
+        end
+
+        # Two candidates or more - Second tie break
+        # Cars that require options with higher utilization rates should enter.
+        if length(candidates) > 1
+            # Compute the utilization rate of each options
+            utilization_rate = Array{Float64,1}(UndefInitializer(),instance.nb_HPRC)
+            for j in 1:instance.nb_HPRC
+                utilization_rate[j] = ( (rv[j] - rpi[j])/len ) / ( instance.RC_p[j] / instance.RC_q[j] )
             end
+            tie_break = zeros(length(candidates))
+            for i in 1:length(candidates)
+                for j in 1:instance.nb_HPRC
+                    tie_break[i] += instance.RC_flag[candidates[i],j] * utilization_rate[j]
+                end
+            end
+
+            # Compute the new candidate list
+            candidates = filter_on_max_criterion(candidates, tie_break)
         end
 
-        # If |candidates| =/= 1 : DOUBLE TIE BREAK
-        #
-        # It may happen that ties still remain after the application of
-        # the above tie breaking criterion. In this case, a second criterion
-        # is added, favoring cars that require options with higher
-        # utilization rates. The utilization rates are dynamically computed,
-        # in the sense that they are updated whenever a new car is added at
-        # the end of the sequence.
+        # Two candidates or more - LPRC criterion
+        # Choose the car that induces the smallest number of new violations of LPRC
+        # when inserted at the end of the current partial sequence.
         if length(candidates) > 1
-            #TODO
+            # Compute the number of violations caused by each car
+            nb_new_violation = zeros(Int, length(candidates))
+            for ind in 1:length(candidates)
+                c = candidates[ind]
+                for j in (instance.nb_HPRC+1):(instance.nb_LPRC+instance.nb_LPRC)
+                    if instance.RC_flag[c, j]
+                        for i in ((position - instance.RC_q[j])+1):position
+                            if solution.M1[j, i] >= instance.RC_p[j]
+                                nb_new_violation[c] = nb_new_violation[c] + 1
```

Comment:
`c` is the car, you want its index `ind` here

Reply:
you're also right